tests/dataconverter/test_helpers.py

```diff
@@ -365,23 +468,23 @@ def fixture_filled_test_data(template, tmp_path):
         pytest.param(TEMPLATE, "", id="valid-data-dict"),
         pytest.param(
             remove_from_dict(TEMPLATE, "/ENTRY[my_entry]/required_group/description"),
-            "The required group, /ENTRY[entry]/required_group, hasn't been supplied.",
+            "The required group, /ENTRY/required_group, hasn't been supplied.",
             id="missing-empty-yet-required-group",
         ),
         pytest.param(
             remove_from_dict(TEMPLATE, "/ENTRY[my_entry]/required_group2/description"),
-            "The required group, /ENTRY[entry]/required_group2, hasn't been supplied.",
+            "The required group, /ENTRY/required_group2, hasn't been supplied.",
             id="missing-empty-yet-required-group2",
         ),
         pytest.param(
             alter_dict(
                 remove_from_dict(
                     TEMPLATE, "/ENTRY[my_entry]/required_group/description"
                 ),
-                "/ENTRY[my_entry]/required_group",
+                "/ENTRY[entry]/required_group",
                 None,
             ),
-            "The required group, /ENTRY[entry]/required_group, hasn't been supplied.",
+            "The required group, /ENTRY/required_group, hasn't been supplied.",
```

Comment:
Should not this report follow consequently either 
- Template style (so we keep [entry]) - I would vote for that, or  
- the actually missing hdf group (so /entry/...) - problem is that the missing item may have a variadic concept name, or 
- the NeXus concept (so we add the name of the AppDef, too, like /NXtest/ENTRY/...)

Reply:
> Should not this report follow consequently either
> 
> * Template style (so we keep [entry]) - I would vote for that, or

Yes, but it seemed a bit odd to me that we report the generic names if the full concept is missing. But we can certainly change it back to `/ENTRY[entry]/...`. I just did it this way because it was easier from the variadic groups where I only have a concept name (or multiple entries). It also seemed to make more sense to me.
> * the actually missing hdf group (so /entry/...) - problem is that the missing item may have a variadic concept name, or

In case we know in which group we are operating we are reporting this specific group. This results in different error logs here, containing either `/ENTRY[my_entry]/...` or `/ENTRY[entry]/...` based on that.
> * the NeXus concept (so we add the name of the AppDef, too, like /NXtest/ENTRY/...)

Possible as well but normally users know that they used the reader with this particular appdef (as specified by `--nxdl`) so it seems a bit redundant to me.


Generally, all three options are possible from a technical perspective. We just have to decide which one it shall be.

Reply:
I changed it now such that we use `/ENTRY[entry]/...` generic paths everywhere if the concept is variadic or unnamed (i.e. not present).